src/python/WMComponent/RucioInjector/Database/MySQL/GetCompletedBlocks.py

```diff
@@ -86,21 +85,29 @@ def format(self, result):
         # NOTE: We need to rename all the keys to follow the cammelCase standard. And also to comply
         #       with the key names as expected from the rest of the already existing python code
         keyMap = {'blockname': 'blockName',
-                  'name': 'workflowName',
+                  'name': 'workflowNames',
                   'pnn': 'location',
                   'site': 'sites',
                   'path': 'dataset',
                   'create_time': 'blockCreateTime'}
 
-        dictResults = DBFormatter.formatDict(self, result)
-        for record in dictResults:
-            for dbKey, pyKey in keyMap.items():
-                if dbKey == 'site':
-                    sites = record.pop(dbKey)
-                    record[pyKey] = set()
-                    record[pyKey].add(sites)
-                else:
-                    record[pyKey] = record.pop(dbKey)
+        listResults = DBFormatter.formatDict(self, result)
+        dictResults = {}
+        for record in listResults:
+            # Populates results dict and adds all workflows of the same block to a single record
+            blockname = record['blockname']
+            if blockname in dictResults:
+                dictResults[blockname]['workflowNames'].add(record['name'])
+                dictResults[blockname]['sites'].add(record['site'])
```

Comment:
IIUC this line here would just add the site to the set of sites of an already existing block name, skipping the values  of the rest of the columns in the record. I am not sure this is safe. Are we not losing information here?

Reply:
the other columns are block name, location, dataset, and creation time. All of those are the same for each block, regardless of the subscription in the record.

Reply:
As above, put this little detail at a comment in the code, please.